examples/ecommerce-netlify-functions/functions/checkout.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+const stripe = require('stripe')(process.env.STRIPE_KEY);
+
+const { Cart, Order, Product } = require('../models');
+
+const handler = async(event) => {
+  try {
+    const stripeProducts = { line_items: [] };
+    let total = 0;
+    for (let i = 0; i < event.body.product.length; i++) {
```

Comment:
Checkout should take in a cart id. Not an array of product/quantity. In the first case, you can assume that the cart is already valid. In your case, you need to validate the data again.